suite/suite.go

```diff
@@ -23,24 +23,49 @@ type Suite struct {
 	*assert.Assertions
 	require *require.Assertions
 	t       *testing.T
+	f       *testing.F
 }
 
 // T retrieves the current *testing.T context.
 func (suite *Suite) T() *testing.T {
 	return suite.t
 }
 
+// F retrieves the current *testing.F context.
+func (suite *Suite) F() *testing.F {
+	return suite.f
+}
+
 // SetT sets the current *testing.T context.
 func (suite *Suite) SetT(t *testing.T) {
 	suite.t = t
 	suite.Assertions = assert.New(t)
 	suite.require = require.New(t)
 }
 
+// SetF sets the current *testing.F context.
+func (suite *Suite) SetF(f *testing.F) {
+	suite.f = f
+	suite.Assertions = assert.New(f)
+	suite.require = require.New(f)
```

Comment:
I think this might be problematic. If we called `suite.SetT(t)` then called `suite.SetF()`, would that not lose the "T" we set in `SetT(t)`?

If so, it seems like that would be the wrong behaviour? Should we not find a way to add an "F" or "T" if the relevant Assertions and requires have already been instantiated?

Reply:
Good point. The current change assumes that you will not have a situation like that. From my current very limited understanding of both testify and [go fuzz tests](https://go.dev/doc/fuzz/), it seems like for having a suite of fuzz tests, you will only have a "top-level" `testing.F` context. 

The [fuzz target](https://go.dev/doc/fuzz/#glos-fuzz-target) has a `testing.T` context, but that's not relevant i think to `testify` as that gets invoked by the go test runner.